dttools/src/jx_sub.c

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright (C) 2021- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "jx_sub.h"
+
+#include <assert.h>
+#include <string.h>
+
+
+static struct jx * jx_sub_call( struct jx *func, struct jx *args, struct jx *ctx) {
+    assert(func);
+    assert(args);
+    assert(jx_istype(args, JX_ARRAY));
+    assert(jx_istype(func, JX_SYMBOL));
+
+    struct jx *left  = jx_copy(func);
+    struct jx *right = NULL;
+
+    if (!strcmp(func->u.symbol_name, "range")    ||
```

Comment:
I think we need to factor this out into a function called something like `jx_is_function_name` that is shared with the corresponding code in `jx_eval` and `jx_function` so that we don't get trapped by changes to the set of functions.  (Which can and do happen)